cpp/src/gandiva/engine.cc

```diff
@@ -274,6 +277,27 @@ Status Engine::LoadPreCompiledIR() {
   return Status::OK();
 }
 
+Status Engine::LoadExternalPreCompiledIR() {
+  auto const& buffers = LLVMExternalIRStore::GetIRBuffers();
+  for (auto const& buffer : buffers) {
+    auto module_or_error = llvm::parseBitcodeFile(buffer->getMemBufferRef(), *context());
+    if (!module_or_error) {
+      std::string str;
+      llvm::raw_string_ostream stream(str);
+      stream << module_or_error.takeError();
+      return Status::CodeGenError("Failed to parse bitcode file, error: " + stream.str());
```

Comment:
Neat, but why isn't there a helper function that automates this error conversion from LLVM to Status?

Reply:
Thanks for pointing it out, I initially want to avoid the duplicated code but forgot to do it later. Now I refactor this piece of code further, as well as the code for module verification and linking to avoid duplicated code here. Please check it out. Thanks.